Kyler/MusicPlayer/app/src/main/java/com/example/kyler/musicplayer/Utils/Helper.java

```diff
@@ -11,14 +11,33 @@
     public static Song getSong(String path){
         MediaMetadataRetriever mmr = new MediaMetadataRetriever();
         mmr.setDataSource(path);
-        String title = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_TITLE);
-        String artist = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ARTIST);
-        String album = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ALBUM);
-        String author = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_AUTHOR);
+        String title = "";
+        String artist = "";
+        String album = "";
+        String author = "";
+        if(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_TITLE) != null){
+            title = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_TITLE);
+        }else{
+            title = "no name";
+        }
+        if(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ARTIST) != null){
+            artist = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ARTIST);
+        }else{
+            artist = "not available";
+        }
+        if( mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ALBUM) != null){
+            album = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ALBUM);
+        }else{
+            album = "not available";
+        }
+        if(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_AUTHOR) != null){
+            author = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_AUTHOR);
+        }else{
+            author = "not available";
+        }
```

Comment:
I follow your advice that create the "fake" information for the file that doesn't have regular information likes title, artist, album ,author.